allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -0,0 +1,305 @@
+        if not os.path.exists(local_dir):
+            os.makedirs(local_dir)
+        if not os.path.isdir(local_dir):
+            raise RuntimeError(f"{local_dir}\n"
+                               "is not a directory")
+
+        bucket_name = bucket_name_from_uri(file_attributes.uri)
+        obj_key = relative_path_from_uri(file_attributes.uri)
+
+        n_iter = 0
+        max_iter = 10  # maximum number of times to try download
+
+        version_id = file_attributes.version_id
+
+        while not self._file_exists(file_attributes):
+            response = self.s3_client.get_object(Bucket=bucket_name,
+                                                 Key=str(obj_key),
+                                                 VersionId=version_id)
+
+            if 'Body' in response:
+                with open(local_path, 'wb') as out_file:
+                    for chunk in response['Body'].iter_chunks():
+                        out_file.write(chunk)
+
+            n_iter += 1
+            if n_iter > max_iter:
+                return False
+        return True
+
+    def data_path(self, file_id) -> pathlib.Path:
+        """
+        Return the local path to a data file, downloading the file
+        if necessary
+
+        Parameters
+        ----------
+        file_id:
+            The unique identifier of the file to be accessed
+
+        Returns
+        -------
+        pathlib.Path
+            The path indicating where the file is stored on the
+            local system
+
+        Raises
+        ------
+        RuntimeError
+            If the file cannot be downloaded
+        """
+        file_attributes = self._manifest.data_file_attributes(file_id)
+        is_valid = self._download_file(file_attributes)
+        if not is_valid:
+            raise RuntimeError("Unable to download file\n"
+                               f"file_id: {file_id}\n"
+                               f"{file_attributes}")
+
+        return file_attributes.local_path
+
+    def metadata_path(self, fname: str) -> pathlib.Path:
+        """
+        Return the local path to a metadata file, downloading the
+        file if necessary
+
+        Parameters
+        ----------
+        fname: str
+            The name of the metadata file to be accessed
+
+        Returns
+        -------
+        pathlib.Path
+            The path indicating where the file is stored on the
+            local system
+
+        Raises
+        ------
+        RuntimeError
+            If the file cannot be downloaded
+        """
+        file_attributes = self._manifest.metadata_file_attributes(fname)
+        is_valid = self._download_file(file_attributes)
+        if not is_valid:
+            raise RuntimeError("Unable to download file\n"
+                               f"file_id: {fname}\n"
+                               f"{file_attributes}")
+
+        return file_attributes.local_path
+
+    def metadata(self, fname: str) -> pd.DataFrame:
```

Comment:
I would expect these getters to start with `get_*`

Reply:
done